samples/csharp_dotnetcore/49.qnamaker-all-features/README.md

```diff
@@ -44,6 +44,31 @@ QnA Maker CLI to deploy the model.
   - Select `QnABot.csproj` file
   - Press `F5` to run the project
 
+##### Microsoft Teams channel group chat fix
+- Goto `Bot/QnABot.cs`
+- Add References
+    ~~~
+    using Microsoft.Bot.Connector;
+    using System.Text.RegularExpressions;
+    ~~~
+- Modify `OnTurnAsync` function as:
+    ~~~
+    public override async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default)
+        {
+            // Teams group chat
+            if (turnContext.Activity.ChannelId.Equals(Channels.Msteams))
```

Comment:
you should use the built in Remove Mention functions: https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Schema/ActivityExtensions.cs#L41

Reply:
in fact do we really need this code snippet in the README at all?

Reply:
This is required as with multi-turn dialog, we cache previous prompt questions, and if any one of these are asked by user, we call qnamaker with corresponding id. But with <at> tag, there is no exact match and it cause score discrepancy.    
Please correct me if I am wrong, the code that you shared is removal from Activity class, which is used for output, so how can we modify incoming queries with this function?